build_tools/configure_ci.py

```diff
@@ -174,37 +180,100 @@ def should_ci_run_given_modified_paths(paths: Optional[Iterable[str]]) -> bool:
 
 
 # --------------------------------------------------------------------------- #
-# Working with pull request labels
+# Matrix creation logic based on PR, push or workflow_dispatch
 # --------------------------------------------------------------------------- #
 
+amdgpu_family_info_matrix = {
+    "gfx94X": {
+        "linux": {"runs-on": "linux-mi300-1gpu-ossci-rocm", "target": "gfx94X-dcgpu"}
+    }
+}
+
 
-def get_pr_labels() -> List[str]:
+def get_pr_labels(args) -> List[str]:
     """Gets a list of labels applied to a pull request."""
-    labels = json.loads(os.environ.get("PR_LABELS", "[]"))
+    labels = json.loads(args.get("pr_labels"))
     return labels
 
 
+def matrix_generator(is_pull_request, is_workflow_dispatch, is_push, args):
+    """Parses and generates build matrix with build requirements"""
+    potential_linux_targets = []
+    potential_windows_targets = []
+
+    # For the specific event trigger, parse linux and windows target information
+    # if the trigger is a workflow_dispatch, parse through the inputs and retrieve the list
+    if is_workflow_dispatch:
+        input_linux_gpu_targets = args.get("input_linux_amdgpu_families")
+        input_windows_gpu_targets = args.get("input_windows_amdgpu_families")
+        translator = str.maketrans(string.punctuation, " " * len(string.punctuation))
+        potential_linux_targets = input_linux_gpu_targets.translate(translator).split()
+        potential_windows_targets = input_windows_gpu_targets.translate(
+            translator
+        ).split()
```

Comment:
What is this "translator" code trying to do? Should at least add a comment showing an example before/after and explaining why (e.g. "sanitize the string to be a valid job name which contains these characters instead of these other characters")